File: javascript/grains/grains.js
'use strict';

export const square = (n) => {
  // throw new Error('Remove this statement and implement this function');

  if (n < 1 || n > 64) {
    throw new Error('square must be between 1 and 64');
  }

  return Math.pow(2, n - 1);
};

export const total = (n) => {
  // throw new Error('Remove this statement and implement this function');
  return Math.pow(2, 64) - 1;
};

File: zig/grains/grains.zig
const std = @import("std");
const math = std.math;

pub const ChessboardError = error{IndexOutOfBounds};

const min_size = 1;
const max_size = 64;

pub fn square(index: isize) ChessboardError!u64 {
    if (index < min_size or index > max_size) {
        return ChessboardError.IndexOutOfBounds;
    }

    const i = @intCast(u64, index) - 1;

    return math.pow(u64, 2, i);
}

pub fn total() u64 {
    return math.maxInt(u64);
}
